### PR TITLE
feat(process-files): allow for broken templates when listing custom components

### PR DESCRIPTION
**Pull request by pkkummermo**

### CODE REVIEW

**Comment by pkkummermo** [comment]
This should fix https://github.com/aurelia/vscode-extension/issues/125 and partially https://github.com/aurelia/vscode-extension/issues/115 as well help with some of the issues in the comment https://github.com/aurelia/vscode-extension/issues/116#issuecomment-603392114

File: src/server/FileParser/ProcessFiles.ts [hunk 1]
```diff
@@ -129,112 +131,118 @@ function processFile(sourceFile: SourceFile) {
   return getCodeInformation(sourceFile);
 }
 
-function processClassDeclaration(node: Node) {
+interface IProcessedClassDeclaration {
```
Comment (hiaux0) [inline src/server/FileParser/ProcessFiles.ts:140]: Your pref to have that one below `function processFile`?

File: src/server/FileParser/ProcessFiles.ts [hunk 2]
```diff
@@ -129,112 +131,118 @@ function processFile(sourceFile: SourceFile) {
...
+  properties: Properties;
+  methods: Methods;
+  name?: string;
+  modifiers?: string[];
+}
+
+function processClassDeclaration(classDeclaration: ClassDeclaration): IProcessedClassDeclaration {
```
Comment (hiaux0) [inline src/server/FileParser/ProcessFiles.ts:147]:
```suggestion
function processClassDeclaration(classDeclarationNode: ClassDeclaration): IProcessedClassDeclaration {
```

Think I want the extra `node` mentioned in the var here

File: src/server/FileParser/ProcessFiles.ts (hunk 1 above)
Comment (pkkummermo) [reply src/server/FileParser/ProcessFiles.ts:140]: I'd actually like to have it on top of the file :) It was just placed there as I drilled downward and that was the first time it became apparent I needed it. I'll push a new change.

Comment (hiaux0) [reply src/server/FileParser/ProcessFiles.ts]: I'm fine with having rough changes.
An initial comment would be nice to indicate that :)

**Pull request merged**